### PR TITLE
Add MergeSort class and Main class

### PR DESCRIPTION
**Pull request by sapozhnykov**

### CODE REVIEW

File: src/com/luxoft/java/course/sapozhnykov/homework/algorithm/sorting/merge/MergeSort.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package com.luxoft.java.course.sapozhnykov.homework.algorithm.sorting.merge;
+
+import java.util.Arrays;
+
+public class MergeSort {
+    public static int[] sort(int[] arr){
+        if(arr.length < 2)
```
Comment (aleksashyn) [inline src/com/luxoft/java/course/sapozhnykov/homework/algorithm/sorting/merge/MergeSort.java]: Please, use {}

File: src/com/luxoft/java/course/sapozhnykov/homework/algorithm/sorting/merge/MergeSort.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            return arr;
+        int middle = arr.length / 2;
+        int[] leftHalf = Arrays.copyOfRange(arr, 0, middle);
+        int[] rightHalf = Arrays.copyOfRange(arr, middle, arr.length);
+        return merge(sort(leftHalf), sort(rightHalf));
+    }
+
+    public static int[] merge(int[] leftHalf, int[] rightHalf){
+        int n = leftHalf.length + rightHalf.length;
```
Comment (aleksashyn) [inline src/com/luxoft/java/course/sapozhnykov/homework/algorithm/sorting/merge/MergeSort.java]: Please, use more human readable names

File: src/com/luxoft/java/course/sapozhnykov/homework/algorithm/sorting/merge/MergeSort.java (hunk 1 above)
Comment (sapozhnykov) [reply src/com/luxoft/java/course/sapozhnykov/homework/algorithm/sorting/merge/MergeSort.java]: Already fixed

File: src/com/luxoft/java/course/sapozhnykov/homework/algorithm/sorting/merge/MergeSort.java (hunk 2 above)
Comment (sapozhnykov) [reply src/com/luxoft/java/course/sapozhnykov/homework/algorithm/sorting/merge/MergeSort.java]: Already fixed

**Review by aleksashyn — APPROVED** [review]

**Pull request merged**